predict.py

```diff
@@ -149,12 +155,17 @@ def predict(
             if stop_sequences:
                 stop_sequences = stop_sequences.split(",")
 
-            if USE_SYSTEM_PROMPT:
-                prompt = (
-                    prompt.strip("\n").removeprefix(B_INST).removesuffix(E_INST).strip()
-                )
-                prompt = PROMPT_TEMPLATE.format(
-                    system_prompt=system_prompt.strip(), instruction=prompt.strip()
+            if USE_SYSTEM_PROMPT and prompt_template:
+                if system_prompt.strip():
+                    if B_SYS not in prompt_template:
+                        # mistral doesn't have a SYS token, there's just a space between the system prompt and 
+                        system_prompt = system_prompt + " "
```

Comment:
this is an incredibly sketchy way of doing it and I would *love* to remove it

Reply:
maybe something like this?

```python
if USE_SYSTEM_PROMPT:
    if not system_prompt.strip():
        print(
            "Note: You have passed an empty system prompt."
        )
    prompt = prompt_template.format(system_prompt=system_prompt, prompt=prompt)
else:
    prompt = prompt_template.format(prompt=prompt)
```